### DISCUSSION TITLE
Any Way to Tell Commands are Executing? + Jumbled Logs

### DISCUSSION THREAD

**Discussion by dbareis** [hluk/CopyQ#2015] [Q&A]
Hi,

Two related issues

**[1]** 
I checked the API and if there is way to tell if a command (or what commands) are executing then I can't see it.  I want to execute one hotkey's command followed by another when the first has completed, I'd like the 2nd to wait for completion.

**[2]** 
Obviously COPYQ seems to handle commands running concurrently yet it provides no way to distringwish the threads/commands that I can see. It just mizes the lines together making the log almost impossible to decypher.

These two lines are from different commands:
```
CopyQ Note [2022-06-07 13:01:23.509] <Server-33144>: EXTRACT MATCH #1, USING: /Funeral director\n(.*)\nTenure expiry date/
CopyQ Note [2022-06-07 13:01:23.913] <Server-33144>: Genealogy Creation (v2022-06-07b) - STARTING
```

**Comment by hluk** [comment]
Commands started by the app itself (from menu, using command or automatically on clipboard change) are listed in Process Manager (Ctrl+Shift+Z).

But it would be really handy to list all processes from command line (or script) and perhaps even be able to terminate specific one.

ad 2. Depending on how you launch the command you could **print to application log** using `console.warn()`, `console.error()` `console.log()` (requires DEBUG logs enabled and the command started from the app, not separately from command line) etc. instead of using `serverLog()`. The process ID shows up in the logs along each line, for example:

```
Warning [2022-06-14 11:51:44.970] <Client-547871>: [js] QtWarning: TESTING...
Warning [2022-06-14 11:51:44.970] <Client-547871>:    another line
Warning [2022-06-14 11:51:44.970] <Client-547871>:    yet another line (eval code:1, %entry)
```

**Comment by dbareis** [comment]
I will look into the process manager as a solution in future, for now I have managed to work around my issue.

The hot key mentioned doesn't work but I found it in the menu (no hotkey mentioned there!).

The console.log sounds potentially interesting (but not for this issue) but I can't get it to output anything in a command started via a copyq global hotkey.

Before trying I stopped copyq and then:

In a command line:
`set COPYQ_LOG_LEVEL=DEBUG`

Started COPYQ

Used Process Explorer to confirm that the process picked up the environment variable.
